### PR TITLE
Remove legacy server support code

### PR DESCRIPTION
**Pull request by jdpgrailsdev**
## What
* Remove unused legacy server-related code as part of cleanup for the migration to Micronaut
* fix https://github.com/airbytehq/airbyte/issues/21599
* Copy of https://github.com/airbytehq/airbyte/pull/23213

## How
* Remove legacy server code
* Remote legacy server code dependencies (Jetty, etc)
* Utilize request logger provided by Micronaut/Netty instead of custom logger
* Clean up dependency catalog/dependency usage
* Remove deprecated task to ensure that there is an unique output directory per task for the API code generation.

## Recommended reading order
1. `airibyte-api/build.gradle`
2. `airbyte-server/application.yaml`
3. `airbyte-server/build.gradle`
4. `deps.toml`

**N.B.** This PR should not be merged until https://github.com/airbytehq/airbyte/pull/23124 has been merged and integration into Cloud.

Below is an example of the built-in access log format:

```2023-02-17 16:58:01 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:00 +0000] "POST /api/v1/workspaces/list HTTP/1.0" 200 315
2023-02-17 16:58:01 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:01 +0000] "POST /api/v1/workspaces/get HTTP/1.0" 200 298
2023-02-17 16:58:07 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:07 +0000] "POST /api/v1/workspaces/update HTTP/1.0" 200 390
2023-02-17 16:58:07 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:07 +0000] "POST /api/v1/web_backend/connections/list HTTP/1.0" 200 18
2023-02-17 16:58:07 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:07 +0000] "POST /api/v1/destination_definitions/list_for_workspace HTTP/1.0" 200 119635
2023-02-17 16:58:07 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:07 +0000] "POST /api/v1/source_definitions/list_for_workspace HTTP/1.0" 200 1290066
2023-02-17 16:58:08 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:07 +0000] "POST /api/v1/web_backend/check_updates HTTP/1.0" 200 50
2023-02-17 16:58:21 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:21 +0000] "GET /api/v1/health HTTP/1.0" 200 18
```

The format can be changed via configuration if we want to use a slightly different logging format.

## Can this PR be safely reverted / rolled back?
- [X] YES 💚
- [ ] NO ❌

## 🚨 User Impact 🚨
None

### CODE REVIEW

**Review by cgardens — APPROVED** [review]

File: airbyte-commons-server/build.gradle [hunk 1]
```diff
@@ -43,26 +44,16 @@ dependencies {
     implementation libs.airbyte.protocol
     implementation project(':airbyte-persistence:job-persistence')
 
-    implementation 'com.github.slugify:slugify:2.4'
-    implementation 'commons-cli:commons-cli:1.4'
+    implementation libs.slugify
```
Comment (davinchia) [inline airbyte-commons-server/build.gradle:47]: nice!

File: airbyte-server/build.gradle [hunk 2]
```diff
@@ -49,26 +45,17 @@ dependencies {
...
     implementation project(':airbyte-persistence:job-persistence')
 
-    implementation 'com.github.slugify:slugify:2.4'
-    implementation 'commons-cli:commons-cli:1.4'
+    implementation libs.slugify
     implementation libs.temporal.sdk
-    implementation 'org.apache.cxf:cxf-core:3.4.2'
-    implementation 'org.eclipse.jetty:jetty-server:9.4.31.v20200723'
-    implementation 'org.eclipse.jetty:jetty-servlet:9.4.31.v20200723'
-    implementation 'org.glassfish.jaxb:jaxb-runtime:3.0.2'
```
Comment (davinchia) [inline airbyte-server/build.gradle:58]: all the removed deps were jersey related?

File: airbyte-server/src/main/java/io/airbyte/server/CorsFilter.java [hunk 3]
```diff
@@ -1,29 +0,0 @@
...
-import com.google.common.net.HttpHeaders;
-import java.util.Map;
-import javax.ws.rs.container.ContainerRequestContext;
-import javax.ws.rs.container.ContainerResponseContext;
-import javax.ws.rs.container.ContainerResponseFilter;
-
-// https://medium.com/@Leejjon_net/how-to-allow-cross-origin-requests-in-a-jax-rs-microservice-d2a6aa2df484
-public class CorsFilter implements ContainerResponseFilter {
-
-  public static final ImmutableMap<String, String> MAP = ImmutableMap.of(
```
Comment (davinchia) [inline airbyte-server/src/main/java/io/airbyte/server/CorsFilter.java:17]: for my knowledge - how do we turn this on for Micronaut?

File: airbyte-server/src/main/resources/application.yml [hunk 4]
```diff
@@ -12,6 +12,10 @@ micronaut:
         web:
           allowedOrigins:
             - ^.*$
+    netty:
+      access-logger:
+        enabled: ${HTTP_ACCESS_LOG_ENABLED:true}
```
Comment (davinchia) [inline airbyte-server/src/main/resources/application.yml:17]: what does this do?

**Review by davinchia — APPROVED** [review]
always love 💣 code @jdpgrailsdev 

super appreciate cleaning up the dep files as part of this!

File: airbyte-server/build.gradle (hunk 2 above)
Comment (jdpgrailsdev) [reply airbyte-server/build.gradle:58]: @davinchia Correct.  Some were not even used by the code -- I assume some old stuff that were not removed when some code was previously removed.

File: airbyte-server/src/main/java/io/airbyte/server/CorsFilter.java (hunk 3 above)
Comment (jdpgrailsdev) [reply airbyte-server/src/main/java/io/airbyte/server/CorsFilter.java:17]: https://docs.micronaut.io/latest/guide/#cors

File: airbyte-server/src/main/resources/application.yml (hunk 4 above)
Comment (jdpgrailsdev) [reply airbyte-server/src/main/resources/application.yml:17]: Enables request logging.  You can see an example of what the output looks like by default in the description of this PR:

```
2023-02-17 16:58:01 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:01 +0000] "POST /api/v1/workspaces/get HTTP/1.0" 200 298
2023-02-17 16:58:07 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:07 +0000] "POST /api/v1/workspaces/update HTTP/1.0" 200 390
2023-02-17 16:58:07 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:07 +0000] "POST /api/v1/web_backend/connections/list HTTP/1.0" 200 18
2023-02-17 16:58:07 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:07 +0000] "POST /api/v1/destination_definitions/list_for_workspace HTTP/1.0" 200 119635
2023-02-17 16:58:07 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:07 +0000] "POST /api/v1/source_definitions/list_for_workspace HTTP/1.0" 200 1290066
2023-02-17 16:58:08 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:07 +0000] "POST /api/v1/web_backend/check_updates HTTP/1.0" 200 50
2023-02-17 16:58:21 INFO i.m.h.s.n.h.a.e.AccessLog(log):125 - airbyte-webapp.airbyte_airbyte_internal - - [17/Feb/2023:16:58:21 +0000] "GET /api/v1/health HTTP/1.0" 200 18
```

**Review by benmoriceau — APPROVED** [review]

**Comment by octavia-approvington** [comment]
Send it! <br/> rocket socks

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**